Library/Formula/sha2.rb

```diff
@@ -0,0 +1,20 @@
+class Sha2 < Formula
+  homepage "http://www.aarongifford.com/computers/sha.html"
+  url "http://www.aarongifford.com/computers/sha2-1.0.1.tgz"
+  sha1 "ffbc69e0b0ab47f70e61aeb14fa16ba2b74dc728"
+
+  option "without-check", "Skip compile-time tests"
+
+  def install
+    system ENV.cc, "-O2", "-DSHA2_UNROLL_TRANSFORM", "-Wall", "-o", "sha2", "sha2prog.c", "sha2.c"
```

Comment:
If particularly interested:

*With no declared ` -O2 ` and a declared ` -DSHA2_UNROLL_TRANSFORM `:

```
TEST SET SIZE: 16.000 MB
SHA-256 average: 0.3022 (52.937 MBps)
SHA-256 best:    0.2930 (54.602 MBps)
SHA-384 average: 0.1926 (83.087 MBps)
SHA-384 best:    0.1881 (85.062 MBps)
SHA-512 average: 0.1928 (82.981 MBps)
SHA-512 best:    0.1864 (85.845 MBps)
```

*With no declared ` -O2 ` and no declared ` -DSHA2_UNROLL_TRANSFORM `:

```
TEST SET SIZE: 16.000 MB
SHA-256 average: 0.2977 (53.740 MBps)
SHA-256 best:    0.2917 (54.845 MBps)
SHA-384 average: 0.1935 (82.685 MBps)
SHA-384 best:    0.1874 (85.390 MBps)
SHA-512 average: 0.1924 (83.156 MBps)
SHA-512 best:    0.1877 (85.263 MBps)
```

*With a declared ` -O2 ` and a declared ` -DSHA2_UNROLL_TRANSFORM `:

```
TEST SET SIZE: 16.000 MB
SHA-256 average: 0.2964 (53.986 MBps)
SHA-256 best:    0.2905 (55.070 MBps)
SHA-384 average: 0.1931 (82.838 MBps)
SHA-384 best:    0.1879 (85.142 MBps)
SHA-512 average: 0.1921 (83.307 MBps)
SHA-512 best:    0.1856 (86.208 MBps)
```